### ISSUE TITLE
Problem with certificate validity in Vivaldi on macOS

### ISSUE THREAD

**Issue by nulcow** [Rehike/Rehike#293]
### Rehike version

0.7

### Operating system

Darwin 21.6.0 (macOS 12.6.5 21G531 arm64)

### PHP version

Whatever the latest XAMPP for macOS (8.2.4) uses. I couldn't tell otherwise.

### What's going on?

In the Vivaldi settings, I clicked the option to manage security certificates. This opened Keychain Access.app, where I then added the certificate to the system-level keychain. It said the certificate was untrusted at first, but I went into the settings and changed it and now Keychain Access says the certificate is "marked as trusted for all users".
Now, when I navigate to youtube.com, I get the Chrome cert error screen. I clicked to make sure it was using the right certificate, and this was what I saw:
<img width="1765" alt="Screenshot 2023-05-26 at 11 22 46 AM" src="https://github.com/Rehike/Rehike/assets/88356560/5ea93417-3f00-4746-8671-19770123cb9c">
It appears to be using the certificate signed by Taniko Yamamoto for youtube.com, so it should be okay. I also set up everything else I needed. Is there something I'm doing wrong?

**Comment by ephemeralViolette** [comment]
Does it work correctly with other browsers than Vivaldi? Since they seem to use Keychain Access on macOS rather than their own systems (like they do on Windows), I doubt it, but it's worth checking.

I also recommend you experiment with different certificates or whatever. I was having issues with Chrome recently, so I did a bunch of random things which got it working. Notably:
- I imported the Apache certificate into the browser. No idea if this is what made it work or not.
- I reimported the browser certificate and let the browser choose where to put it rather than manually putting it as an authority.

(Although I think my issue there was related to incognito mode not updating along with the regular profile.)

It's been a while since I used macOS at all, so unfortunately this is the best advice I can give at the moment.
